src/coreclr/tools/Common/Internal/Runtime/MethodTable.Constants.cs

```diff
@@ -57,7 +57,10 @@ internal enum EETypeFlags : ushort
         /// </summary>
         OptionalFieldsFlag = 0x0100,
 
-        // Unused = 0x0200,
+        /// <summary>
+        /// The _usComponentSize is a number (not holding FlagsEx).
+        /// </summary>
+        HasComponentSizeFlag = 0x0200,
```

Comment:
We should match the tricks that we do for this in MethodTable. It is `0x80000000` so that the JIT can optimize this as sign check. Also, the flags are fetched as dword so that it can use the smaller instructions (on x64).

Reply:
Like - merge the flags and extended into one `uint32` and move the `HasComponentSize` to the sign bit?

or just move the bit?

Reply:
Check the codegen for `go_through_object` for CoreCLR and make sure that it is as good.

Reply:
Right, we want `mT->HasComponentSize()` in the following just be a sign check. 
(now it probably loads a short into a 32bit register and does a bit test)

```c
inline size_t my_get_size (Object* ob)
{
    MethodTable* mT = header(ob)->GetMethodTable();

    return (mT->GetBaseSize() +
            (mT->HasComponentSize() ?
             ((size_t)((CObjectHeader*)ob)->GetNumComponents() * mT->RawGetComponentSize()) : 0));
}
```

The perf here is typically gated by the cache misses when fetching the mT pointer - we are looking at headers of semi-random objects, and that is often a miss.  

It would not hurt if codegen for `HasComponentSize` is tighter though.